crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -109,9 +112,59 @@ impl JsonRpcHandler {
     /// starknet_getTransactionByHash
     pub(crate) async fn get_transaction_by_hash(
         &self,
-        _transaction_hash: TransactionHashHex,
+        transaction_hash: TransactionHashHex,
     ) -> RpcResult<TransactionWithType> {
-        Err(error::ApiError::TransactionNotFound)
+        let starknet = self.api.starknet.read().await;
+        let transaction_to_map = starknet.transactions.get(&transaction_hash.0).unwrap();
+        let transaction_type;
+        let transaction_data: Transaction = match transaction_to_map.inner.clone() {
+            starknet_core::transactions::Transaction::Declare(declare_v1) => {
+                transaction_type = TransactionType::Declare;
+
+                // Is case of a reverted transaction we need to set class_hash.
+                // It's just the first iteration of get_transaction_by_hash() code so probably this
+                // will change soon with v0.4.0 RPC spec adoption.
+                let class_hash_to_set = if declare_v1.class_hash.is_some() {
+                    FeltHex(declare_v1.class_hash.unwrap())
+                } else {
+                    FeltHex(Felt::from(0))
+                };
```

Comment:
You can use `.unwrap_or(...)`: https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.unwrap_or

Reply:
Fixed https://github.com/0xSpaceShard/starknet-devnet-rs/commit/6ba5b10fbe83722fd4ede7cbdcc79fd02604969e I used unwrap_or_default()